contracts/RandomBeaconHistory.cdc

```diff
@@ -0,0 +1,96 @@
+/// RandomBeaconHistory (FLIP 123)
+///
+/// This contract stores the history of random sources generated by the Flow network. The defined Heartbeat resource is
+/// updated by the Flow Service Account at the end of every block with that block's source of randomness.
+///
+/// While the source values are safely generated by the Random Beacon (non-predictable, unbiasable, verifiable) and transmitted into the execution 
+/// environment via the committing transaction, using the raw values from this contract does not guarantee non-revertible
+/// randomness. The Hearbeat is intended to be used in conjunction with a
+/// commit-reveal mechanism to provide an onchain source of non-revertible randomness.
+// It is also recommended to use the source values with a pseudo-random number 
+// generator (PRNG) to generate an arbitrary-long sequence of random values. 
+// 
+// For usage of randomness where result abortion is not an issue, it is recommended
+// to use the Cadence built-in function `revertibleRandom`, which is also based on 
+// the safe Random Beacon. 
+///
+/// Read the full FLIP here: https://github.com/onflow/flips/pull/123
+///
+access(all) contract RandomBeaconHistory {
+
+    /// The height at which the first source of randomness was recorded
+    access(contract) var initHeight: UInt64?
```

Comment:
In case we decide to change this in the future (to clear some history or something), I feel like a more appropriate name would be `lowestHeight`